app.py

```diff
@@ -0,0 +1,369 @@
+        return jsonify({'message': 'invalid input'})
+
+    if phone_number:
+        user.phone_number = phone_number
+    if first_name:
+        user.first_name = first_name
+    if last_name:
+       user.last_name = last_name
+    update_details()
+    return jsonify({'message': 'User details are updated successfully'})  
+
+
+@app.route('/delete',methods=['DELETE'])
+@jwt_required()
+def delete_user():
+    user_id = get_jwt_identity()
+    user = User.query.filter_by(id=user_id).first()
+    if user:
+       delete(user)
+       return jsonify({'message': 'data deleted'})
+    else:
+        return jsonify({'error': 'User cannot be deleted'})
+
+
+@app.route('/change_password', methods=['POST'])
+@jwt_required()
+def change_password():
+    user_id = get_jwt_identity()
+    data = request.json
+    password = data.get('password')
+    new_password = data.get('new_password')
+    confirm_new_password = data.get('confirm_new_password')
+    
+    user = User.query.filter_by(id=user_id).first()
+    if not user:
+        return jsonify({'message': 'User not found'})
+
+    if not bcrypt.check_password_hash(user.password, password):
+        return jsonify({'message': 'Incorrect password'})
+
+    if not new_password:
+        return jsonify({'message': 'New password not provided'})
+    if not confirm_new_password:
+        return jsonify({'message': 'Confirm_New password not provided'})
+    if confirm_new_password != new_password:
+        return jsonify({'message': 'Confirm_New password and new password field not match'})
+    else:
+
+        hashed_password = bcrypt.generate_password_hash(new_password).decode('utf-8')
+        user.password = hashed_password
+        db.session.commit()
+        return jsonify({'message': 'Password changed successfully'})
+
+
+@app.route('/forget_password', methods=['POST'])
+def forget_password():
+    data = request.json
+    email_id=data.get('email_id')
+    user = User.query.filter_by(email_id=email_id).first()
+    if not user:
+        return jsonify({'message': 'User not found'})
+    else:  
+         encoded_email_id = quote_plus(email_id) 
+         reset_link = f'http://127.0.0.1:5000/reset_password/{encoded_email_id}'
+         msg = Message( 
+                'Hello', 
+                sender = os.getenv('MAIL_USERNAME'),
+                recipients = [email_id]) 
+         msg.body =  f'Hello,\n\nYour reset link is/ {reset_link}'
+         mail.send(msg) 
+
+         return jsonify({'message': 'The Mail is Send.'}) 
+
+
+@app.route('/reset_password/<encoded_email_id>', methods=['POST'])
+def reset_password(encoded_email_id):
+
+    email_id = unquote_plus(encoded_email_id)  
+    data = request.json
+    new_password = data.get('new_password')
+    confirm_new_password = data.get('confirm_new_password')
+   
+    user = User.query.filter_by(email_id=email_id).first()
+    if not user:
+        return jsonify({'message': 'Invalid email ID'})
+    if confirm_new_password != new_password:
```

Comment:
Where are the error messages displayed when the new password and confirm new password fields are left blank?